File: 4sum.cpp
#include<bits/stdc++.h>

using namespace std;

int findSumzero(vector<int> v1, vector<int> v2, vector<int> v3, vector<int> v4){
    int answer = 0;
    map<int, int> tracker;
    for(int i = 0; i < v1.size(); i++){
        for(int j = 0; j < v2.size(); j++){
            int sum = v1[i] + v2[j];
            tracker[sum] += 1;
        }
    }
    for(int k = 0; k < v3.size(); k++){
        for(int l = 0; l < v4.size(); l++){
            int negsum = - (v3[k] + v4[l]);
            answer = answer + tracker[negsum];
        }
    }
    return answer;
}

int main(){
    vector<int> v1, v2, v3, v4;
    v1 = { 1, 2 };
    v2 = { -2, -1 };
    v3 = { -1, 2 };
    v4 = { 0, 2 };
    int number = findSumzero(v1, v2, v3, v4);
    cout<<"The number of possible answers is "<<number<<endl;    
    return 0;
}

File: 4sum.py
def findSumzero(arr1, arr2, arr3, arr4):
    answer = 0
    m = {}
    for i in range(len(arr1)):
        for j in range(len(arr2)):
            sumple = arr1[i] + arr2[j]
            if sumple not in m:    
                m[sumple] = 0
            m[sumple] += 1
    for k in range(len(arr3)):
        for l in range(len(arr4)):
            sumple1 = - ( arr3[k] + arr4[l] )
            if sumple1 in m:
                answer += m[sumple1]
    return answer
    

arr1 = [ 1, 2 ]
arr2 = [-2, -1]
arr3 = [-1, 2 ]
arr4 = [0, 2]

answer = findSumzero(arr1, arr2, arr3, arr4)
print('The number of possible ways are {}'.format(answer))